engine/apps/api/views/alert_receive_channel.py

```diff
@@ -159,7 +159,14 @@ def get_queryset(self, eager=True, ignore_filtering_by_available_teams=False):
         if not ignore_filtering_by_available_teams:
             queryset = queryset.filter(*self.available_teams_lookup_args).distinct()
 
-        queryset = self.filter_by_labels(queryset)
+        # filter labels
+        labelQuery = self.request.query_params.getlist("label", [])
+        kvPairs = parse_label_query(labelQuery)
+        for key, value in kvPairs:
+            queryset = queryset.filter(
+                labels__key_id=key,
+                labels__value_id=value,
+            ).distinct()
```

Comment:
nit: maybe add `distinct()` add the end of the method once? I know it won't change the SQL but still maybe it'd be just a little easier to read.